### PR TITLE
Update README defaults and config

### LINKED ISSUE #1020: Update configuration section with new default values
**What happened?**
I was reading through the README.md to learn and try out kube-prometheus when I noticed that the [Configuration section](https://github.com/prometheus-operator/kube-prometheus/tree/main#configuration) seem to describe the old configuration (release-0.7 and earlier). It has the old `_config` instead of `values` for example.

**Did you expect to see some different?**
I was expecting the new configuration defaults (which I'm not sure how they really look like) as hinted in the [migration guide](https://github.com/prometheus-operator/kube-prometheus/blob/main/docs/migration-guide.md).

**How to reproduce it (as minimally and precisely as possible)**:
Check the [configuration section of the README.md](https://github.com/prometheus-operator/kube-prometheus/tree/main#configuration).

**Environment**

* Prometheus Operator version:

    `v0.46.0`
    

* Kubernetes version information:

    ```
    Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
    Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-21T01:11:42Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
    ```
    

* Kubernetes cluster kind:

    Kubernetes in docker: `kind create cluster`

### PR DESCRIPTION
**Pull request by lentzi90**
Fixes #1020. I'm not at all sure about this and really just looking for feedback at this stage.

So I understand that the defaults come from [this](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/main.libsonnet#L14-L108), but since quite a lot is done through variables and the old default section didn't have any of those I'm not sure what to do. Should I "resolve" those variables so that the true value is written out? This is what I have tried to do now for a few of them, but there are many more and I'm not sure if all of them should be included.

Let me know what you think! :slightly_smiling_face:

### CODE REVIEW

**Comment by lentzi90** [comment]
@paulfantom any input on this?

File: README.md [hunk 1]
```diff
@@ -332,71 +332,98 @@ Jsonnet has the concept of hidden fields. These are fields, that are not going t
...
+      versions: {
+        alertmanager: error 'must provide version',
+        blackboxExporter: error 'must provide version',
+        grafana: error 'must provide version',
+        kubeStateMetrics: '1.9.8',
+        nodeExporter: error 'must provide version',
+        prometheus: error 'must provide version',
+        prometheusAdapter: error 'must provide version',
+        prometheusOperator: error 'must provide version',
+      } + (import 'versions.json'),
```
Comment (paulfantom) [inline README.md:352]: Maybe we should add another sub-section about version numbers we are using and link to versions.json file there? That file is meant to be automatically updated and now it holds all component versions.

**Review by paulfantom — COMMENTED** [review]
> Should I "resolve" those variables so that the true value is written out?

I think it would be better to point to a file with variable values. For example, we are pretty frequently updating component versions and including those in README can lead to documentation staleness.

With recent refactoring and making code a bit more readable, I think we might even go one step further and rewrite first example in `Configuration` paragraph to only link to `main.libsonnet` and explain that main configuration is done in `values` map.

File: README.md (hunk 1 above)
Comment (lentzi90) [reply README.md]: Alright I made a second attempt now. :slightly_smiling_face: Removed all the defaults and just linked to `main.libsonnet` and `versions.json` instead. Let me know what you think.

**Comment by paulfantom** [comment]
Awesome! Thanks! :100:

**Pull request merged**